### ISSUE TITLE
[Content Patcher] random token problem

### ISSUE THREAD

**Issue by dalbiblue** [Pathoschild/LookupAnything#529]
I recently created a mode to marry a wizard. It is the Romanceable Rasmodius mode I edited.
I used a random token in this mode.
However, after the token's marriage to the wizard, I found that the token was unable to kiss him and the ambassador was not printing properly. What is wrong?
This is what i wrote. ↓

{
            "Action": "Load",
            "Target": "Characters/Dialogue/Wizard",
            "FromFile": "assets/Characters/Dialogue/wizard-{{random: 0, 1, 2, 3, 4}}.json"
        },

{
            "Action": "Load",
            "Target": "Characters/Dialogue/MarriageDialogueWizard",
            "FromFile": "assets/Characters/Dialogue/MarriageDialogueWizard.json"
        },

**Comment by Pathoschild** [comment]
Can you do this to provide more info?

1. Try it again.
2. In the SMAPI console window, enter this command:  
   `patch summary`
3. Upload your SMAPI log to [smapi.io/log](https://smapi.io/log/) (see instructions on that page).
4. Post the link here.

**Comment by dalbiblue** [comment]
https://smapi.io/log/a2d738d02fbc4c1794cde9a6947428aa
Thank you. I will send you my mod if you need.

**Comment by Pathoschild** [comment]
It looks like you didn't load the save. Can you try talking to the wizard (so you should see their dialogue), then run the command and post the log?

**Comment by dalbiblue** [comment]
https://smapi.io/log/b747e115c6fe46c1bc5f13df0fbabe86

sorry, my mistake!

**Comment by dalbiblue** [comment]
https://smapi.io/log/cd993ddabcfb4f609e3cf962f8d6056c

i'm sorry, my mistake!

**Comment by Pathoschild** [comment]
Both `Characters/Dialogue/Wizard` and `Characters/Dialogue/MarriageDialogueWizard` seem to be loaded and applied correctly. Neither of those patches should have any effect on whether you can kiss him, but I'm not sure what you mean by "the ambassador was not printing properly"; could you rephrase?

**Comment by dalbiblue** [comment]
Even when the wizard is outside the house, he speaks indoors line. Sorry my bad English.
And, I can kiss him by subtracting the random token.

**Comment by Pathoschild** [comment]
The game only updates dialogue when the day starts, so unfortunately you can't have indoor/outdoor dialogue using only Content Patcher conditions. (It is possible with [location dialogue](https://stardewvalleywiki.com/Modding:Dialogue#Location_dialogue) though.)

Can you post a zip of the mod and [your save folder](https://stardewvalleywiki.com/Saves)? I'll take a look at the kissing issue.

**Comment by dalbiblue** [comment]
[save file.zip](https://github.com/Pathoschild/StardewMods/files/4328018/save.file.zip)
[romantic wizard mod.zip](https://github.com/Pathoschild/StardewMods/files/4328019/romantic.wizard.mod.zip)

Thank you! Incidentally, I can kiss sometimes. But most can't.

**Comment by dalbiblue** [comment]
You may already know, according to the report of users who used this mode, if you can have an additional conversation at 11:00, you can kiss after 11:00 even if you did not kiss in the morning. I hope this report helps. Thank you!

**Comment by Pathoschild** [comment]
Sorry for the slow response! Work has been busy.

When I loaded your save, the wizard was outside and I was able to kiss him right away after talking to him. Does it work for you if you temporarily remove all mods except Content Patcher and the Romanceable Rasmodius content pack?

**Comment by dalbiblue** [comment]
Thanks for taking a look at my mods!
As you advised, I tried removing other mods and playing romantic wizard mod. But still couldn't kiss the wizard.  I was able to kiss on the first day, but I can't kiss him after the next day. Can you take a look again? 
Thank you very much for taking the time out of your busy schedule.

**Comment by Pathoschild** [comment]
@dalbiblue I can't reproduce it, so we'll need to narrow it down. Does it still happen if you remove all patches except `Data\NPCDispositions` (so he's still marriageable), that dialogue patch, and the minimum of other patches required for the mod to work? Can you post a zip of the smaller content pack if so?

**Comment by dalbiblue** [comment]
I narrow it down and attach it. The only remaining are Character sprite, Dialogue, MarriageDialogue, and NPCDispositions. I tried again with this miniature version but still couldn't kiss the next day.
I hope this problem solves well.
[Romantic Wizard_narrow ver.zip](https://github.com/Pathoschild/StardewMods/files/4399546/Romantic.Wizard_narrow.ver.zip)

**Comment by Pathoschild** [comment]
I reproduced it with that version, but it's not related to Content Patcher.

The issue happens when the game adds a marriage dialogue key that doesn't exist in the file, like `spouseRoom_Wizard`. The game won't let you kiss him because he has dialogue to say, but then won't show the dialogue because the key doesn't exist.

The next [Lookup Anything](https://www.nexusmods.com/stardewvalley/mods/541) update lets you see which marriage dialogue key is still queued in the datamining fields; feel free to ping me [on Discord](https://stardewvalleywiki.com/Modding:Help#Creating_mods) if you'd like a preview build.
> ![](https://user-images.githubusercontent.com/230581/77972963-d91c0180-72c0-11ea-9643-c0bb3633bf47.png)
